tester/runner.go

```diff
@@ -139,6 +140,30 @@ func (r *Runner) SetCompiler(compiler *ast.Compiler) *Runner {
 	return r
 }
 
+type Builtin struct {
+	Decl *ast.Builtin
+	Func func(*rego.Rego)
+}
+
+// Warning: If you set a new compiler after this method, it overrides the compiler with the registered custom builtins
+func (r *Runner) SetBuiltins(builtinsList []*Builtin) *Runner {
+	capabilities := ast.CapabilitiesForThisVersion()
+	for _, builtin := range builtinsList {
+		capabilities.Builtins = append(capabilities.Builtins, builtin.Decl)
+		r.builtins = append(r.builtins, builtin.Func)
+	}
+
+	if r.compiler == nil {
```

Comment:
Could we not just add this logic to the `runTests` call instead? It already has this:

```go
if r.compiler == nil {
    r.compiler = ast.NewCompiler().
	WithEnablePrintStatements(enablePrintStatements)
}
```
Seems to me we could extand that to:
```go
if r.compiler == nil {
    r.compiler = ast.NewCompiler().
	WithEnablePrintStatements(enablePrintStatements).
        WithBuiltins(r.builtins)
}
```
or something like that

Reply:
Great idea! Thanks to this comment the code is became more robust and clean, in fact now the warning comment is useless.